jetty-util/src/main/java/org/eclipse/jetty/util/resource/URLResource.java

```diff
@@ -272,9 +272,7 @@ public Resource addPath(String path)
         throws IOException
     {
         if (path == null)
-            return null;
-
-        path = URIUtil.canonicalPath(path);
+            throw new MalformedURLException("null path");
```

Comment:
Is this change from returning null  to MalformedURLException tested?

Reply:
yes - The main resource classes do not return null and few calls to addPath ever expect a null return.
So this is just making it more uniform.